core/src/main/java/com/linecorp/armeria/client/retry/RetryConfigBuilder.java

```diff
@@ -0,0 +1,147 @@
+
+    @Nullable
+    private final RetryRuleWithContent<T> retryRuleWithContent;
+
+    /**
+     * Returns a {@link RetryConfigBuilder} with this {@link RetryRule}.
+     */
+    RetryConfigBuilder(RetryRule retryRule) {
+        this.retryRule = requireNonNull(retryRule, "retryRule");
+        retryRuleWithContent = null;
+        maxContentLength = 0;
+    }
+
+    /**
+     * Returns a {@link RetryConfigBuilder} with this {@link RetryRuleWithContent}.
+     */
+    RetryConfigBuilder(RetryRuleWithContent<T> retryRuleWithContent) {
+        retryRule = null;
+        this.retryRuleWithContent = requireNonNull(retryRuleWithContent, "retryRuleWithContent");
+        maxContentLength = Integer.MAX_VALUE;
+    }
+
+    /**
+     * Sets the maxContentLength to be used with a {@link RetryRuleWithContent}.
+     */
+    public RetryConfigBuilder<T> maxContentLength(int maxContentLength) {
+        checkArgument(maxContentLength > 0,
+                      "maxContentLength: %s (expected: > 0)", maxContentLength);
+        this.maxContentLength = maxContentLength;
+        return this;
+    }
+
+    /**
+     * Sets maxTotalAttempts.
+     */
+    public RetryConfigBuilder<T> maxTotalAttempts(int maxTotalAttempts) {
+        checkArgument(
+                maxTotalAttempts > 0,
+                "maxTotalAttempts: %s (expected: > 0)",
+                maxTotalAttempts);
+        this.maxTotalAttempts = maxTotalAttempts;
+        return this;
+    }
+
+    /**
+     * Sets responseTimeoutMillisForEachAttempt.
+     */
+    public RetryConfigBuilder<T> responseTimeoutMillisForEachAttempt(long responseTimeoutMillisForEachAttempt) {
+        checkArgument(
+                responseTimeoutMillisForEachAttempt >= 0,
+                "responseTimeoutMillisForEachAttempt: %s (expected: >= 0)",
+                responseTimeoutMillisForEachAttempt);
+        this.responseTimeoutMillisForEachAttempt = responseTimeoutMillisForEachAttempt;
+        return this;
+    }
+
+    /**
+     * Sets responseTimeoutMillisForEachAttempt by converting responseTimeoutForEachAttempt to millis.
+     */
+    public RetryConfigBuilder<T> responseTimeoutForEachAttempt(Duration responseTimeoutMillisForEachAttempt) {
+        final long millis =
+                requireNonNull(responseTimeoutMillisForEachAttempt, "responseTimeoutMillisForEachAttempt")
+                        .toMillis();
+        checkArgument(
+                millis >= 0,
+                "responseTimeoutForEachAttempt.toMillis(): %s (expected: >= 0)",
+                millis);
+        this.responseTimeoutMillisForEachAttempt = millis;
+        return this;
+    }
+
+    /**
+     * Builds a {@link RetryConfig} from this builder's values and returns it.
+     */
+    public RetryConfig<T> build() {
+        if (retryRule != null) {
+            return new RetryConfig<>(retryRule, maxTotalAttempts, responseTimeoutMillisForEachAttempt);
+        }
+        if (retryRuleWithContent != null) {
```

Comment:
How about just adding `assert retryRuleWithContent != null` and remove if condition?

Reply:
Done